WebApp/Utils/MSALPerUserMemoryTokenCache.cs

```diff
@@ -74,26 +69,23 @@ public MSALPerUserMemoryTokenCache(ITokenCache tokenCache, ClaimsPrincipal user)
         {
             Initialize(tokenCache, user);
         }
-        
+
         /// <summary>Initializes the cache instance</summary>
         /// <param name="tokenCache">The ITokenCache passed through the constructor</param>
         /// <param name="user">The signed-in user for whom the cache needs to be established..</param>
         private void Initialize(ITokenCache tokenCache, ClaimsPrincipal user)
         {
             SignedInUser = user;
```

Comment:
How are we certain that the user is the right one? 
Should we really keep the user in the cache? rather than re-finding the account from the `ClaimsPrincipal.Current`?
@kalyankrishna1 ; what do you think?

Reply:
The user is the right one because every time we need to get a token, we call: `MsalAppBuilder.BuildConfidentialClientApplication()` and this method uses the **ClaimsPrincipal.Current** as shown here: 

```csharp
 public static IConfidentialClientApplication BuildConfidentialClientApplication()
 {
      return BuildConfidentialClientApplication(ClaimsPrincipal.Current);
 }
```